argilla/src/argilla/records/_mapping/_mapper.py

```diff
@@ -83,6 +84,24 @@ def __call__(self, data: Dict[str, Any], user_id: Optional[UUID] = None) -> Reco
         metadata = self._map_attributes(data=data, mapping=self.mapping.metadata)
         vectors = self._map_attributes(data=data, mapping=self.mapping.vector)
 
+        unknown_keys = [key for key in data.keys() if key not in self.mapping.keys()]
+        if unknown_keys:
+            warnings.warn(f"Keys {unknown_keys} in data are not present in the mapping and will be ignored.")
+
+        if len([k for k in data if k != self.mapping.id.source]) == 0:
```

Comment:
can we rewrite the condition to:
```suggestion
        if not data:
```
? 

If data have some value there is the other check to raise the error ( Maybe I'm missing some scenarios)

Reply:
Here I'm also checking that the only value is not an id, because that will fail the same as `{}`